City_CRUD_API/router.py

```diff
@@ -0,0 +1,71 @@
+from fastapi import Depends, HTTPException, APIRouter
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from . import schemas, crud
+from dependencies import get_db
+
+router = APIRouter()
+
+
+# Retrieve a list of all cities
+@router.get("/cities/", response_model=list[schemas.CityList])
+async def read_cities(db: AsyncSession = Depends(get_db)):
+    return await crud.get_all_cities(db=db)
+
+
+# Get the details of a specific city by ID
+@router.get("/cities/{city_id}/", response_model=schemas.CityList)
+async def read_city(city_id: int, db: AsyncSession = Depends(get_db)):
+    city = await crud.get_city_by_id(db=db, city_id=city_id)
+    if not city:
+        raise HTTPException(status_code=404, detail="City not found")
+    return city
+
+
+# Create a new city
+@router.post("/cities/", response_model=schemas.CityList)
+async def create_city(
+        city: schemas.CityCreate,
+        db: AsyncSession = Depends(get_db),
+):
+    db_city = await crud.get_city_by_name(db=db, name=city.name)
+
+    if db_city:
+        raise HTTPException(
+            status_code=400,
+            detail="Such name for City already exists"
+        )
+
+    return await crud.create_city(db=db, city=city)
```

Comment:
In `create_city`, the check for an existing city name is good practice. Ensure that the error message is clear and consistent with other error messages in the API.